### PR TITLE
fix(ui): close ten visual-taste and product-sense gaps

### PR DESCRIPTION
**Pull request by zz-plant**
## Summary

Ten independent fixes, all found by running the app and looking at it rather than by reading the code. The diff is wide but shallow.

**The landing page contradicted its own tagline.** "Full-screen visuals" sat over a stage capped at 680px on every desktop size, so a 1080p viewport spent its lower third on empty page ground. Three separate things were shrinking it: that cap, a shell padding that made the workspace taller than the box containing it (an 8px overflow that raised a scrollbar and pushed the "centred" frame off centre), and a document-wide `scrollbar-gutter: stable` reserving 15px down the right edge forever. Home is full-bleed now, like live mode already was.

**Browse gave no visual basis for choosing.** Most of the catalog has no thumbnail, and the fallback was one of seven mood gradients plus the mood as text — so a screen of tiles was the same navy square a dozen times, and the art slot spent itself restating what the caption beside it already said. `PresetArtworkFallback` generates a picture per preset instead: deterministic from an FNV-1a hash of the id, palette anchored to the mood so a kind still reads as a family, drawn as the landing page's own signal-trace motif so the placeholder reads as part of the product. Four SVG paths from integer maths — no canvas, no WebGL, no network, memoized for the virtualized grid.

**Every panel opened by spotlighting its own exit.** The focus trap took the first focusable descendant, which in a panel whose header comes first is the close button — so opening one lit up "dismiss", and screen readers announced "Close, button" in place of the panel's name. `useFocusTrap` takes `initialFocus: 'container'` now, the standard dialog behaviour.

**Panels arrived as generic skeletons for seconds.** Chunks only began downloading once a panel was picked. Opening the stage menu now prefetches what that menu can launch, through a shared `panel-chunks` module that also serves the existing route and idle prewarms. The Suspense fallback is browse-shaped for browse rather than four identical bars, and skeleton fill/sheen became tokens — as raw white-alpha they were invisible against the light theme's cream, so every panel loading in light mode showed a completely blank sheet.

**The transport bar under-served the one action people take constantly.** Previous and next were 32px buttons identical in size and weight to save, mute, full screen and the overflow menu, so the bar offered six equal affordances and no answer to "what do I press?". They are the primary pair now. The preset title was set smaller and lighter than the menu items behind it; the credit sat at 0.68rem, below the size at which its tuned contrast ratio even applies. The bar and its "Controls" handle also both rendered during the bar's 300ms fade, stacking the word Controls on top of the preset title — they are one control in two states now, with a coordinated handoff.

**The loading screen spoke in engineering phases.** "Starting app shell", "Showing controls", "Starting visualizer runtime" describe our architecture, not anything a visitor asked about, and they flicker past in under a second on a warm load. One calm line now, with a slow-load line at 6s that gives a reason. Removed the dismiss X, which tore the loader off and left an unfinished page behind it.

**Hints were placed by rule, not by context.** "Tap a card to play it" fired the moment the route said browse — several seconds before the catalog lands on a cold load, and on the `/discover` entry points. Since a hint marks itself seen the first time it shows, it was spent pointing at an empty sheet and never appeared again once there was something to point at. Hints now carry an anchor and render in the surface they describe rather than in the stage's toast slot on the far side of the screen, the browse hint waits for cards to exist, and it clears when acted on instead of outliving the card you just tapped.

**The audio-source disclosure opened into a different product.** A 2×2 of cards, each led by a tracked uppercase kicker ("NO SETUP", "LIVE SOURCE") paired with a decorative WebGL identicon that at 28px in a card corner read as a status light — together spending the card's whole top edge to say less than its own title, and costing four live WebGL contexts on the landing page. One row of four peers now, each with an icon that says what the source is. The section heading said "YouTube playback" while naming a section containing YouTube, microphone, a file and this tab. The primary button was a lime gradient found nowhere else in the palette, while "Play demo" one disclosure above it was the brand accent.

**Light theme was half-built.** The launch hero pinned its ink to the on-stage palette but not its surfaces, so cards and fields drew cream (`--card-bg`) and then wrote near-white on top: the YouTube field and all four source cards rendered as blank slabs with invisible labels. The home stage frame was separately repainted with a cream gradient, a grey border and a drop shadow, turning a permanently dark full-screen render into a dark rectangle on a pale card — an embedded video, not the product. Surfaces are pinned alongside the inks, the stage is dark in every theme (the rule `tokens.css` already stated), and the active Settings tab is no longer white-on-white.

**The declared design system described a product that does not exist.** The radius block was headed "Kiki: Sharp &amp; Angular Geometry" and topped out at 6px, while the app drew 19 literal pixel radii up to 28px; the type scale was declared once and then bypassed by 41 distinct rem sizes, 13 of them packed inside the 0.6–0.85rem band where the steps are smaller than a pixel. Both are snapped onto real scales — 123 radius declarations and 190 font sizes, maximum delta 2px and 0.1rem — the smallest UI text is raised to an 11px floor, and a new `check:css-scale` guard keeps them there, because a scale is only real if something enforces it.

**Fonts came from a third party and guaranteed a reflow.** IBM Plex loaded from Google Fonts behind a `media="print"` onload swap: that avoided render-blocking, but guaranteed first paint in the system font and a reflow of every glyph after three round trips — most visibly on the 5.5rem wordmark. Self-hosted latin subsets now (one 46KB variable Sans covering 400–700, two 15KB Mono statics), preloaded, zero third-party requests. Weight 600, which the wordmark uses and the old request never loaded, resolves properly rather than being synthesised.

Also fixes `check:duplicate-css`, which keyed `@font-face` on family name alone and so reported any correctly self-hosted family as a duplicate of itself.

## Testing

Before the merge commit, on the original branch point:

- `bun run check` — passed, **2821 tests across 294 files**, exit 0.
- `bun run build` — succeeded; verified the three woff2 files ship to `dist/fonts/` and that `dist/index.html` contains no `fonts.googleapis` reference.
- `bun run scripts/check-css-scale.ts` — the new guard, passes on the migrated stylesheets.

After merging `main` (which brought TS 7, MCP SDK v2, Vite 8.2, husky→lefthook and knip), re-validated on the merged tree:

- `bun run check:quick` — passes.
- `bun run test --profile unit` — **2747 tests across 276 files**, exit 0.
- `bun run test --profile compat` — 25 tests across 4 files, exit 0.
- `bun run check:dead-code` — main's new knip guard reports nothing from this branch's files. It did flag one export of mine, `PanelChunkId`, correctly; that is now internal.

Verified in a real browser (Playwright + `?agent=true`) at 390×844, 1440×900 and 1920×1080, in both colour schemes. Measured rather than eyeballed where it mattered:

- stage frame goes from `[24, 24, 1377, 680]` on a 1440×900 viewport to `[0, 0, 1440, 900]`, with `scrollHeight === innerHeight` (was 908 vs 900, i.e. the scrollbar).
- `captureStats()` sampled twice reports `motionEstimate &gt; 0`, so the full-bleed stage is still rendering.
- `document.fonts` reports the self-hosted faces loaded, `document.fonts.check('600 16px "IBM Plex Sans"')` is true, and the page makes 0 requests to googleapis/gstatic.
- light-theme launch hero surfaces now compute to `rgba(10, 14, 22, 0.6)` instead of `rgb(255, 250, 240)`.
- 20 generated artworks render in a browse grid, all visually distinct.

Three existing tests asserted the exact copy and JSX this change rewrites, and were updated to the new strings. The artwork test now asserts the contract that actually matters and previously had no coverage: the generated art is byte-identical for the same preset id and differs for a sibling id (`glow` vs `glow-2`).

**One pre-existing failure, not from this branch.** `tests/corpus/preset-flash-risk.test.ts` fails locally with `Target page, context or browser has been closed` during navigation. It fails identically on `origin/main` in a clean worktree with none of this branch's changes present, and it passes here when run in isolation. It is browser-backed and skips where Chromium is absent. Flagging it rather than touching it, since it is outside this PR's scope.

## Docs touched

- `docs/GUARDRAILS.md` — regenerated. Documents the new `check:css-scale` guard; the count reads 30 after the merge, since `check:dead-code` arrived with `main`.
- Rationale for each change is in the code as comments, and in the commit messages.

## Review risk checklist

- [x] Null/undefined paths reviewed for changed logic
- [x] Async/lifecycle state transitions reviewed (loading, visibility, audio, teardown)
- [x] Existing shared helper/module checked before introducing duplicate logic
- [x] New visual literals use existing design tokens (or include a new named token)
- [x] Behavior change is covered by tests (or reason provided)

Notes for the reviewer on the riskiest parts:

- The `--text-*` tokens keep the exact values they had. Redefining them would have silently shrunk every current consumer, so new rungs (`--text-2xs`, `--text-xs-plus`, `--text-sm-plus`, `--text-display`) fill the gaps the literals were being written into instead.
- The scale migration was mechanical, so the visual delta is bounded and stated: no radius moved more than 2px and no font size more than 0.1rem.
- Two `border-radius` literals are deliberately off-scale and allowlisted in the guard with reasons — they shape a blurred ambient glow, not a control corner.
- `ShaderIdenticon` no longer has a consumer after the source cards dropped it. Left in place rather than deleted, since it is a general-purpose component and removing it is a separable decision.
- The merge resolved one conflict, in `package.json`: both sides added a script key at the same point in the alphabetical run, and both were kept.

## Quality checklist

- [x] `bun run check:quick`
- [x] `bun run check` (required for JS/TS changes)
- [x] `bun run build` (if build/runtime output changed)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01TRm36bXdSFXwH6DdB7Bi5o

---
_Generated by [Claude Code](https://claude.ai/code/session_01TRm36bXdSFXwH6DdB7Bi5o)_

### CODE REVIEW

File: src/js/frontend/App.tsx [hunk 1]
```diff
@@ -1163,6 +1197,28 @@ function StimsWorkspaceAppShell() {
...
+      hintPresetRef.current = null;
+      return;
+    }
+    if (hintPresetRef.current === null) {
+      // The render that showed the hint. Remember what was playing then;
+      // only a *change* from here is the user acting on the advice.
+      hintPresetRef.current = activePresetId ?? '';
+      return;
+    }
+    if (hintPresetRef.current !== (activePresetId ?? '')) dismissHint();
```
[the remark this answers is not in the document]
Comment (zz-plant) [reply src/js/frontend/App.tsx]: Confirmed and fixed in 22f87e2. You're right that the preset id is the wrong signal — it moves on its own, and because `showHint` persists the id as seen on first display, losing it once loses it for good.

Dismissal now comes from the panel reporting an actual choice rather than from watching what is playing. `BrowseSheetPanel` takes an `onPresetChosen` callback, and every selection site routes through it. Worth noting there were two paths, not one: three sites call `engine.handlePresetSelection` and now go through a `selectPreset` helper, but the grid opens a preset by committing route state through `setPresetRouteState`, so that reports a choice too. The grid is the default view, so a tile click is in fact the primary "tap a card" the hint is about — hooking only the engine calls would have left the main case still broken.

Verified both directions in a browser with the panel open from first paint, inside the hint's 5s window:

| Action | Hint after |
|---|---|
| Engine-driven preset change | still visible |
| Real tile click | dismissed |

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/check-css-scale.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  }
+  return `${pick}${unit}`;
+}
+
+for (const file of CSS_ROOTS.flatMap((r) => walk(r))) {
+  const lines = stripComments(readFileSync(file, 'utf8')).split('\n');
+  lines.forEach((line, i) => {
+    // `border-radius: 12px`, and the shorthand's first value. Percentages
+    // (50% for circles) and var() references are left alone.
+    for (const m of line.matchAll(/border-radius:\s*([0-9.]+)px/g)) {
```
[the remark this answers is not in the document]
Comment (zz-plant) [reply scripts/check-css-scale.ts]: Confirmed and fixed in 22f87e2, including the instance you named. The guard now reads every length in the declaration, not just one after the colon, so it also covers the elliptical half after a slash.

Two things came out of it beyond the pattern change:

**The `SidePanel.module.css` case was a real defect, not just a missed check.** The original was `18px 18px 0 0`; my migration replaced only the leading value, leaving the mobile sheet's two top corners at different sizes. Both are the token now.

**Turning the pattern up found a second one, and a false positive.** `chrome.css` has `calc(var(--ctl-radius) - 3px)` for concentric corners. The literal there is the gap between an inner and outer radius, not a corner value, and the result is on the scale by construction because the token it derives from is — so lengths inside `calc()` are excluded rather than allowlisted case by case.

The guard had no test, which is how the hole shipped. `tests/unit/check-css-scale.test.ts` now runs the real script against fixtures so it cannot drift from what the gate runs. Three of the seven cases are shapes that previously passed:

```
border-radius: var(--radius-xl) 18px 0 0        → now rejected (18px)
border-radius: 10px 10px 0 0 / 10px 10px 0 7px  → now rejected (7px)
border-radius: calc(var(--ctl-radius) - 3px)    → still accepted
```

---
_Generated by [Claude Code](https://claude.ai/code)_
